lints/lint_sub_ca_eku_valid_fields.go

```diff
@@ -0,0 +1,41 @@
+package lints
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+)
+
+type subCAEKUValidFields struct {
+	// Internal data here
+}
+
+func (l *subCAEKUValidFields) Initialize() error {
+	return nil
+}
+
+func (l *subCAEKUValidFields) CheckApplies(c *x509.Certificate) bool {
+	return util.IsSubCA(c) && util.IsExtInCert(c, util.EkuSynOid)
+}
+
+func (l *subCAEKUValidFields) RunTest(c *x509.Certificate) (ResultStruct, error) {
+	for _, ekuValue := range c.ExtKeyUsage {
+		if ekuValue == x509.ExtKeyUsageServerAuth ||
+			ekuValue == x509.ExtKeyUsageClientAuth {
+			continue
+		} else {
+			return ResultStruct{Result: Notice}, nil
```

Comment:
The text says:

> Other values MAY be present.

Which is counter-intuitive, but what the document says. So I think this test is wrong. Given the uselessness of this specific requirement, I'm wondering if it should be a lint at all.